dc/s2n-quic-dc/src/path/secret/map/rehandshake.rs

```diff
@@ -38,20 +39,16 @@ impl RehandshakeState {
     }
 
     pub(super) fn push(&mut self, peer: SocketAddr) {
-        self.queue.push(peer);
+        self.queue.push(SocketAddress::from(peer).to_ipv6_mapped());
     }
 
     pub(super) fn adjust_post_refill(&mut self) {
-        self.queue.sort_unstable();
+        // Sort by hash, and if hashes are the same, by the SocketAddr. We need to include both in
+        // the comparison key to ensure that we find duplicate entries correctly (just by hash
+        // might have different addresses interleave).
+        self.queue
+            .sort_unstable_by_key(|peer| (self.hasher.hash_one(peer), *peer));
```

Comment:
won't this hash the address on every comparison? I wonder if it would be better to use `sort_unstable_by` and check to see if the peers match before hashing? I guess that'll only reduce the hashing operations for duplicates which i'm not sure how often that'll happen.

Reply:
It will, but I think matching addresses are very unlikely in practice -- we should only have that happen transiently after a rehandshake (for ~10 minutes) -- so with 500k peers and a 24 hour period, `(500_000/(60*24)*10)/500_000 * 100 = 0.69` = ~0.7% of peers (3472) should have duplicate entries at any time.

I think in practice this also runs ~once a handshake period (so once every 24 hours) so a bit of extra CPU is not a big deal for the hashing. We could cache them (sort_unstable_by_key_cached) but I think the extra memory spend isn't worth it.